CMakeLists.txt

```
@@ -17,20 +18,36 @@ option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
+set(BOARD_NAME "" CACHE INTERNAL "Enter the board name to compile mlpack for.")
+
+if(CMAKE_CROSSCOMPILING)
+  set(GFORTRAN_LIBRARY "" CACHE STRING "Path for gfortran library.")
+  set(PTHREAD_LIBRARY "" CACHE STRING "Path for pthread library")
+  include(board/flags-config.cmake)
+  if(BOARD_NAME STRGREATER  "")
+    flags(${BOARD_NAME})
+  endif()
+endif()
 
 # Set minimum library version required by mlpack.
 set(ARMADILLO_VERSION "8.400.0")
 set(ENSMALLEN_VERSION "2.10.0")
 set(BOOST_VERSION "1.58")
 set(CEREAL_VERSION "1.1.2")
 
+# If BUILD_SHARED_LIBS is OFF then mlpack library will be build statically
+# In addition, all mlpack CLI bindings are going to be linked statically as
+# well.
 if (WIN32)
   option(BUILD_SHARED_LIBS
       "Compile shared libraries (if OFF, static libraries are compiled)." OFF)
```

```suggestion
      "Compile shared libraries (if OFF, static libraries and binaries are compiled)." OFF)
```
We should update this documentation too, otherwise users may not see it (I doubt that many open `CMakeLists.txt`).